portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,33 +42,45 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     PreparedQuery results = datastore.prepare(query);
 
     List<String> comments = new ArrayList<>();
+    if (quantity == 0){ quantity = 10; } //Default set at max of 10 comments shown upon loading screen
+    int i = 0;
     for (Entity entity : results.asIterable()) {
-      long id = entity.getKey().getId();
-      String message = (String) entity.getProperty("comment");
-      Date timestamp = (Date) entity.getProperty("date");
-
-      comments.add(message + " - " + timestamp);
+      if (i < quantity){ //Limits number of comments added to the page
+        long id = entity.getKey().getId();
+        String message = (String) entity.getProperty("comment");
+        Date timestamp = (Date) entity.getProperty("date");
+        comments.add(message + " - " + timestamp);
+        i++;
+      }else {
+        break; //Exits for-loop once requested number of comments appear
+      }
     }
 
     response.setContentType("application/json");
     String json = new Gson().toJson(comments);
     response.getWriter().println(json);
   }
 
-    @Override
+  @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-    //TODO: get input from form
-
     String newComment = request.getParameter("new-comment");
+    String quant = request.getParameter("quantity");
 
-    //comments.leaveComment(newComment);
-    Entity commentEntity = new Entity("Comment");
-    commentEntity.setProperty("comment", newComment);
-    commentEntity.setProperty("date", new Date());
+    try {
+      quantity = Integer.parseInt(quant);
+    } catch (NumberFormatException e) {
+      quantity = 10; //default set at 10 comments
```

Comment:
It looks like if the user does not specify "quantity" and only submit a "comment", the quantity will be reset to 10. I wonder if there can be an improvement.

Reply:
Yes, I noticed this to when testing but I wasn't sure exactly how to improve this - I'll work on it today!